src/index.js

```diff
@@ -1,46 +1,122 @@
-  new Task('reproduce the design in flutter', 3, false),
-  new Task('create the Api', 4, false),
-
-];
-function generateTodoIndex() {
-  let id = 1;
-  if (tasklist.length > 0) {
-    id = tasklist[tasklist.length - 1].index + 1;
-  }
-  return id;
-}
+const tasklist = new TaskList();
 
 function generateTaskList() {
-  let content = '';
-  tasklist.forEach((task) => {
-    content += task.renderLi();
-  });
-  listContainer.innerHTML = content;
-}
-function refreshTodoList(e) {
-  e.preventDefault();
+  listContainer.innerHTML = tasklist.generateTaskList();
+}
+function refreshTodoList() {
   listContainer.innerHTML = '';
   generateTaskList();
 }
-function addTask() {
+function add() {
   if (taskinput.value !== '') {
-    tasklist.push(new Task(taskinput.value, generateTodoIndex()));
+    tasklist.addTask(taskinput.value);
     generateTaskList();
     taskinput.value = '';
-    localStorage.setItem('taskList', JSON.stringify(this.taskList));
+  }
+}
+function hideComponent(elmntsParam) {
+  const test = elmntsParam.classList;
+  if (!test.contains('hidden_item')) {
+    test.toggle('hidden_item');
+  }
+}
+
+function displayComponent(elmnts) {
+  if (elmnts.isArray) {
+    elmnts.forEach((elmnt) => {
+      const test = elmnt.classList;
+      if (test.contains('hidden_item')) {
+        test.remove('hidden_item');
+      }
+    });
+  } else {
+    const test = elmnts.classList;
+    if (test.contains('hidden_item')) {
+      test.remove('hidden_item');
+    }
+  }
+}
+function displayLineThroughComponent(elmntsParam) {
+  const test = elmntsParam.classList;
+  if (!test.contains('line_through')) {
+    test.toggle('line_through');
+  }
+}
+
+function removeLineThroughComponent(elmnts) {
+  if (elmnts.isArray) {
+    elmnts.forEach((elmnt) => {
+      const test = elmnt.classList;
+      if (test.contains('line_through')) {
+        test.remove('line_through');
+      }
+    });
+  } else {
+    const test = elmnts.classList;
+    if (test.contains('line_through')) {
+      test.remove('line_through');
+    }
   }
 }
 
+function completeTask(index) {
+  hideComponent(document.querySelector(`.uncomplete${index}`));
+  displayComponent(document.querySelector(`.complete${index}`));
+  hideComponent(document.querySelector(`.details${index}`));
+  displayComponent(document.querySelector(`.delete${index}`));
+  displayLineThroughComponent(document.querySelector(`.input${index}`));
+  // console.log(tasklist.localData[index-1].getCompleted());
+  tasklist.localData[index - 1].setCompleted(true);
+}
+
+function uncompleteTask(index) {
+  hideComponent(document.querySelector(`.complete${index}`));
+  displayComponent(document.querySelector(`.uncomplete${index}`));
+  hideComponent(document.querySelector(`.delete${index}`));
+  displayComponent(document.querySelector(`.details${index}`));
+  removeLineThroughComponent(document.querySelector(`.input${index}`));
+  tasklist.localData[index - 1].setCompleted(false);
+}
```

Comment:
- [ ] Kindly make sure the user is able to `delete` a task without depending on the `checkbox` button status. Right now, a **user** can only delete a **task or item** if only the `checkbox` button is **checked** but I think deleting a task will not be dependent on the  `checkbox` button status. Make sure you fix that.

![a_todo](https://user-images.githubusercontent.com/83042742/193180048-1cd56293-4547-46b5-a665-2f5b7bb8b16c.gif)